File: cosmos/src/main/java/com/planbtech/cosmos/model/entites/Address.java
package com.planbtech.cosmos.model.entites;

import com.planbtech.cosmos.model.Interfaces.IAddress;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.GenerationType;


@Entity
@Table(name = "addresses")
public class Address implements IAddress {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String street;
    private String city;
    private String state;
    private String CEP;
    private Integer number;


    @Override
    public String getStreet() {
        return this.street;
    }

    @Override
    public String getCity() {
        return this.city;
    }

    @Override
    public String getState() {
        return this.state;
    }

    @Override
    public String getCEP() {
        return this.CEP;
    }

    @Override
    public Integer getNumber() {
        return this.number;
    }

    @Override
    public void setStreet(String street) {
        this.street = street;
    }

    @Override
    public void setCity(String city) {
        this.city = city;
    }

    @Override
    public void setState(String state) {
        this.state = state;
    }

    @Override
    public void setCEP(String cep) {
        this.CEP = cep;
    }

    @Override
    public void setNumber(Integer number) {
        this.number = number;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Address address)) return false;

        return getId() != null ? getId().equals(address.getId()) : address.getId() == null;
    }

    @Override
    public int hashCode() {
        return getId() != null ? getId().hashCode() : 0;
    }
}

File: cosmos/ProConnect/src/app/Class/address.ts
import { IAddress } from "../interfaces/iaddress";
import { IState } from "../interfaces/istate";
import { State } from "./state";

export class Address implements IAddress{

  state: IState = new State();
  city: string = '';
  street: string = '';
  CEP: string = '';
  number: number = 0;

  getCEP(): string {  return this.CEP;  }
  setCEP(CEP: string): void {  this.CEP = CEP;  }

  getNumber(): number {  return this.number;  }
  setNumber(number: number): void {  this.number = number;  }

  getCity(): string {  return this.city;  }
  setCity(city: string): void {  this.city = city;  }

  getStreet(): string {  return this.street;  }
  setStreet(street: string): void {  this.street = street;  }

  getState(): IState {  return this.state;  }
  setState(state: IState): void {  this.state = state;  }

  verificaEndereco():boolean{
    if(this.CEP.length == 8 && this.number > 0 && this.city != "" && this.street != "" && this.state.stateName != ""){
      return true;
    }
    return false;
  }

}
